examples/security-token/assets/controller.py

```diff
@@ -0,0 +1,222 @@
+    issue_token = Seq([
+        assetReserve,
+        Assert(And(
+			Txn.application_args.length() == Int(1),
+			# Txn.assets.length() == Int(1), [TEALv3]
+            Gtxn[1].type_enum() == TxnType.AssetTransfer,
+            Gtxn[1].xfer_asset() == App.globalGet(token_id), # verify if token index is correct
+
+            # Issue should only be done by token reserve
+            assetReserve.hasValue(),
+			Gtxn[0].sender() == assetReserve.value(),
+            Gtxn[1].asset_sender() == assetReserve.value(),
+
+            # only allow issue if token is not killed
+            App.globalGet(var_is_killed) == Int(0)
+        )),
+        Return(Int(1))
+    ])
+
+    # kill a token, after verifying transaction set is_killed to true(Int(1))
+    assetManager = AssetParam.manager(Int(0))
+    kill_token = Seq([
+        assetManager, # load asset_manager of Txn.ForeignAssets[0]
+        Assert(And(
+			Txn.application_args.length() == Int(1),
+			# Txn.assets.length() == Int(1), [TEALv3],
+            # Txn.assets[0] == App.globalGet(token_id), [TEALv3] # verify if token index is correct
+            Txn.type_enum() == TxnType.ApplicationCall,
+
+            # Only asset manager can kill the token
+            assetManager.hasValue(),
+			Txn.sender() == assetManager.value(),
+        )),
+
+        # finally set is_killed to true
+        App.globalPut(var_is_killed, Int(1)),
+        Return(Int(1))
+    ])
+
+    add_new_permission = Seq([
+        assetManager,
+        Assert(And(
+            Txn.application_args.length() == Int(3),
+            # Txn.assets[0] == App.globalGet(token_id), [TEALv3]
+
+            assetManager.hasValue(),
+			Txn.sender() == assetManager.value(),
+        )),
+
+        # Update total rules counter (= 1 in this case) as we only have a single
+        # permissions contract
+        App.globalPut(var_total, App.globalGet(var_total) + Int(1)),
+
+        # Add permissions(rules) smart contract config in global state (id, manager)
+        App.globalPut(permission_id, Btoi(Txn.application_args[1])),
+        App.globalPut(permissions_manager, Txn.application_args[2]),
+        Return(Int(1))
+    ])
+
+    # Token manager can update permissions manager to a different address
+    change_permissions_manager = Seq([
+        assetManager,
+        Assert(And(
+            Txn.application_args.length() == Int(1),
+            # Txn.assets[0] == App.globalGet(token_id), [TEALv3]
+
+            assetManager.hasValue(),
+			Txn.sender() == assetManager.value(),
+        )),
+
+        # update permissions manager address to the first account address passed in appAccounts
+        App.globalPut(permissions_manager, Txn.accounts[1]),
+        Return(Int(1))
+    ])
+
+	# check properties of txGroup passed
+    group_tx_checks = And(
+        # Ensure 3 basic calls + 1 rules contract call is present in group
+        Global.group_size() == Add(Int(3), App.globalGet(var_total)),
```

Comment:
what happens if var_total is > 1 ?
here we are checking four transactions.

Reply:
if var_total = 2 then transactions = 5 (1 more to call the new rules contract). But in this version we only have single rules contract support.